src/Mod/Fem/Gui/Command.cpp

```diff
@@ -1129,10 +1132,191 @@ bool CmdFemCreateNodesSet::isActive()
 
 //================================================================================================
 //================================================================================================
+
+/* start of Erase Element bit */
+DEF_STD_CMD_A(CmdFemDefineElementsSet);
+
+void DefineElementsCallback(void *ud, SoEventCallback *n)
+{
+    Fem::FemAnalysis* Analysis;
+
+//    Base::Console().Warning("DefineElemensCall\n");
+
+    if (getConstraintPrerequisits(&Analysis))
+        return;
+
+    // show the wait cursor because this could take quite some time
+    Gui::WaitCursor wc;
+
+    // When this callback function is invoked we must in either case leave the edit mode
+//    Gui::View3DInventorViewer* view = reinterpret_cast<Gui::View3DInventorViewer*>(n->getUserData());
+    Gui::View3DInventorViewer* view = static_cast<Gui::View3DInventorViewer*>(n->getUserData());
+    view->setEditing(false);
+    view->removeEventCallback(SoMouseButtonEvent::getClassTypeId(), DefineElementsCallback, ud);
+    n->setHandled();
+
+    Gui::SelectionRole role;
+    std::vector<SbVec2f> clPoly = view->getGLPolygon(&role);
+    if (clPoly.size() < 3)
+        return;
+    if (clPoly.front() != clPoly.back())
+        clPoly.push_back(clPoly.front());
+
+    SoCamera* cam = view->getSoRenderManager()->getCamera();
+    SbViewVolume vv = cam->getViewVolume();
+    Gui::ViewVolumeProjection proj(vv);
+    Base::Polygon2d polygon;
+    for (std::vector<SbVec2f>::const_iterator it = clPoly.begin(); it != clPoly.end(); ++it)
+        polygon.Add(Base::Vector2d((*it)[0], (*it)[1]));
+
+    std::vector<App::DocumentObject*> docObj = Gui::Selection().getObjectsOfType(Fem::FemMeshObject::getClassTypeId());
+    if (docObj.size() != 1)
+        return;
+
+//    const SMESHDS_Mesh* data = const_cast<SMESH_Mesh*>(static_cast<Fem::FemMeshObject*>(docObj[0])->FemMesh.getValue().getSMesh())->GetMeshDS();
+    const SMESHDS_Mesh* data = static_cast<Fem::FemMeshObject*>(docObj[0])->FemMesh.getValue().getSMesh()->GetMeshDS();
+
+    SMDS_NodeIteratorPtr aNodeIter = data->nodesIterator();
+    Base::Vector3f pt2d;
+    std::set<int> IntSet;
+
+    while (aNodeIter->more()) {
+        const SMDS_MeshNode* aNode = aNodeIter->next();
+        Base::Vector3f vec(aNode->X(), aNode->Y(), aNode->Z());
+        pt2d = proj(vec);
+        if (polygon.Contains(Base::Vector2d(pt2d.x, pt2d.y)))
+            IntSet.insert(aNode->GetID());
+    }
+
+    std::stringstream  set;
+
+    set << "[";
+    for (std::set<int>::const_iterator it = IntSet.begin(); it != IntSet.end(); ++it)
+        if (it == IntSet.begin())
+            set << *it;
+        else
+            set << "," << *it;
+    set << "]";
+
+//    Gui::Command::openCommand("Place robot");
```

Comment:
robot? In case you copied code from somewhere else, please review what is actually needed and omit not applicable code.
The name "robot" is not clear to me in the context of deleting mesh nodes.